enumerables.rb

```diff
@@ -1,88 +1,100 @@
+# frozen_string_literal: true
+
+# enumerable methods
 module Enumerable
   def my_each
     arr = self
+    if block_given?
       for i in arr
         yield(i)
+        puts i
       end
+    else
+      arr.to_enum(:my_each)
+    end
   end
 
   def my_each_with_index
     arr = self
-      for i in arr
+    if block_give?
+      arr.my_each do |i|
         v = arr.find_index(i)
         yield(i, v)
       end
+    else
+      arr.to_enum(:my_each_with_index)
+    end
   end
 
   def my_select
     arr = self
     result = []
-    for i in arr
-      if  yield i
-        result.push(i)
-        end
+    arr.each do |i|
+      result.push(i) if yield i
     end
     result
   end
 
-  def my_all(argument = nil)
+  def my_all?(arg = nil)
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false unless yield(i)
+        return false unless yield(i)
       end
     else
-      my_each { |i| return false if i.nil? || i == false }
+      my_each { |i| return false if !i.is_a? arg }
     end
     true
   end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all? == array.all? # false
  ```

* [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```ruby
    words = %w[dog door rod blade]
    p words.my_all?(/d/) == words.all?(/d/) # false
  ```
* [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```ruby
    array = [3,3,3,3,3]
    p array.my_all?(3) == array.all?(3) # false
  ```